tests/tizen_capi/unittest_tizen_capi.cc

```diff
@@ -488,6 +488,31 @@ file_cmp (const gchar * f1, const gchar * f2)
   return cmp;
 }
 
+/**
+ * @brief Wait until the change in pipeline status is done
+ * @return ML_ERROR_NONE success, ML_ERROR_UNKNOWN if failed, ML_ERROR_TIMED_OUT if timeout happens.
+ */
+int
+waitPipelineStateChange (ml_pipeline_h handle, ml_pipeline_state_e state,
+    guint timeout_ms)
+{
+  int status = ML_ERROR_UNKNOWN;
+  guint counter = 0;
+  ml_pipeline_state_e cur_state = ML_PIPELINE_STATE_NULL;
+
+  do {
+    status = ml_pipeline_get_state (handle, &cur_state);
+    EXPECT_EQ (status, ML_ERROR_NONE);
+    if (cur_state == ML_PIPELINE_STATE_UNKNOWN)
+      return ML_ERROR_UNKNOWN;
+    if (cur_state == state)
+      return ML_ERROR_NONE;
+    g_usleep (10000);
+  } while ((timeout_ms / 10) > counter++);
+
+  return ML_ERROR_TIMED_OUT;
```

Comment:
How about check the state before returning timed_out? currently, if timeout_ms is less than 10ms, it always returns timed out.

Reply:
Understand!
How about change to
```
  } while ((timeout_ms / 10) >= counter++);
```

Reply:
sound good :)

Reply:
Updated.